Source/WebCore/platform/network/mac/UTIUtilities.mm

```diff
@@ -146,8 +146,14 @@ bool isDeclaredUTI(const String& UTI)
 void setImageSourceAllowableTypes(const Vector<String>& supportedImageTypes)
 {
 #if HAVE(CGIMAGESOURCE_WITH_SET_ALLOWABLE_TYPES)
-    auto allowableTypes = createNSArray(supportedImageTypes);
-    CGImageSourceSetAllowableTypes((__bridge CFArrayRef)allowableTypes.get());
+    // A WebPage might be reinitialized. So restrict ImageIO to the default and
+    // the additional supported image formats only once.
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [supportedImageTypes] {
+        auto allowableTypes = createNSArray(supportedImageTypes);
+        auto status = CGImageSourceSetAllowableTypes((__bridge CFArrayRef)allowableTypes.get());
+        RELEASE_ASSERT_WITH_MESSAGE(supportedImageTypes.isEmpty() || status == noErr, "ImageIO could not install all the supported image decoders.");
```

Comment:
I think "could not install all the supported image decoders" is the wrong message. I doubt `CGImageSourceSetAllowableTypes` actually loads a bunch of decoders; it probably just stores a list somewhere. It would be better if the message just said that CGImageSourceSetAllowableTypes() returned error NNN.